chiselFrontend/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -55,7 +55,18 @@ private[chisel3] class IdGen {
   }
 }
 
-private[chisel3] trait HasId {
+/** Public API to access Node/Signal names.
+  * currently, the node's name, the full path name, and references to its parent Module and component.
+  * These are only valid once the design has been elaborated, and should not be used during its construction.
+  */
+trait SignalId {
+  def signalName: String
+  def pathName(seperator: String = "."): String
+  def parentPathName(seperator: String = "."): String
```

Comment:
Is separator ever changed at runtime?  Can SignalId just have a member like

    private def separator = "."

instead of the default argument to pathName/parentPathName?